server.go

```diff
@@ -77,6 +77,20 @@ func Listen(conn net.PacketConn, config *Config) (Listener, error) {
 	}, nil
 }
 
+func populateConfig(config *Config) *Config {
+	versions := config.Versions
+	if len(versions) == 0 {
+		versions = make([]protocol.VersionNumber, len(protocol.SupportedVersions))
+		copy(versions, protocol.SupportedVersions)
+	}
+
+	return &Config{
+		TLSConfig: config.TLSConfig,
```

Comment:
You can copy a tls.Config using https://golang.org/pkg/crypto/tls/#Config.Clone.

It seems net/tls copies its config when it makes changes, so I'd prefer to do that too. However it doesn't protect against races if the user changes stuff in the config after passing it (the clone is shallow), so I don't feel like we have to.